### PR TITLE
Support for Stax

### PR DESCRIPTION
**Pull request by agrojean-ledger**
## Add Stax compatibility
- Update app for latest zxlib support,
- Remove obsolete nanos SDK gitmodule dependency,
- Use proper gitmodule for zxlib dependency (remove old zxlib version that was versioned as a directory).

## Update tests
- Update tests for compatibility with latest Zemu version.
- Commit golden snapshots of all supported devices (lns,lnx,lnsp,stax).
- Remove babel, yarn config (not used anymore),
- Update package.json (newer modules versions), add package-lock.json,
- Remove obsolete .gitignore files (use only root .gitignore instead),
- Remove snapshots-tmp directories.

## Add CI
- Add rule enforcer,
- Add build reusable workflow,
- Add Zemu test job for all devices.

### CODE REVIEW

File: app/src/json/json_parser.h
Comment (agrojean-ledger) [inline app/src/json/json_parser.h:38]: Required otherwise .bss does not fit in RAM.

File: app/src/tx_display.c
Comment (agrojean-ledger) [inline app/src/tx_display.c:394]: Maybe someone has a better fix for the dangling reference of the local `tmp_val` variable in `parser_tx_obj` ?

File: .github/workflows/ci-workflow.yml
Comment (lpascal-ledger) [inline .github/workflows/ci-workflow.yml]: Maybe we should be able to refer to a stable version of the workflow now?

File: app/src/addr.h
Comment (lpascal-ledger) [inline app/src/addr.h:2]: Zondax changed its company legal form?

**Review by lpascal-ledger — APPROVED** [review]
What I understand from these changes is that most of the work is done in the ZXlib, and all needing to be done on top of it is integrating the correct Makefile flags and C guards, and adapting the CI/tests for Stax, am I right?
LGTM if I am/

**Comment by agrojean-ledger** [comment]
> What I understand from these changes is that most of the work is done in the ZXlib, and all needing to be done on top of it is integrating the correct Makefile flags and C guards, and adapting the CI/tests for Stax, am I right? LGTM if I am/

Yes you are absolutely right @lpascal-ledger . The "heavy lifting" of the GUI logic is totally hidden in zondax's ZXlib. I have updated the code to work with the latest version of this lib and fix a few issues here and there but that's about all I needed to do. 

Most painful was adapting the tests, updating package versions etc etc.

File: app/src/addr.h
Comment (agrojean-ledger) [reply app/src/addr.h:2]: It seems indeed. I copied this one from another zondax app...

File: .github/workflows/ci-workflow.yml
Comment (agrojean-ledger) [reply .github/workflows/ci-workflow.yml]: We can probably (not definetly, need to check). I'll update that and make sure everything runs smoothly before merge.

**Pull request merged**
